compiler/infra/ForwardList.hpp

```diff
@@ -0,0 +1,684 @@
+#ifndef FORWARD_LIST_HPP
```

Comment:
Previous extensions of std structures in OMR used a lower-case name matching the std equivalent (e.g., infra/deque.hpp).  Please rename this file to `forward_list.hpp` for consistency with that precedent.

`TRlist.hpp` is yet another way of naming an extending class that I'm not too comfortable with, and I don't think that naming is the precedent that should be used here.